src/core/ddsc/tests/psmx.c

```diff
@@ -1264,6 +1326,378 @@ CU_Test (ddsc_psmx, basic)
+        cfg.psmx_instances->cfg.library = ddsrt_strdup("psmx_iox");
+        cfg.psmx_instances->cfg.config = ddsrt_strdup(config_str); // config including a locator
+        cfg.psmx_instances->cfg.priority.isdefault = 1;
+        cfg.psmx_instances->cfg.priority.value = 1000000;
+        cfg.psmx_instances->next = NULL;
+        domain = dds_create_domain_with_rawconfig(domainId, &cfg);
+        config_psmx_elm_free(cfg.psmx_instances);
+      }
+      CU_ASSERT_FATAL(domain > 0);
+      CU_ASSERT_FATAL(domain_has_psmx_instance_name(domain, "CycloneDDS-IOX-PSMX"));
+      dds_entity_t participant = dds_create_participant(domainId, NULL, NULL);
+      CU_ASSERT_FATAL(participant > 0);
+      dds_entity_t writer1, reader1, writer2, reader2;
+      {
+        char topicname[100];
+        dds_qos_t* qos = dds_create_qos();
+        dds_qset_history(qos, DDS_HISTORY_KEEP_LAST, sample_cnt);
+
+        create_unique_topic_name("shared_memory", topicname, sizeof(topicname));
+        dds_entity_t topic1 = dds_create_topic(participant, &SC_Model_desc, topicname, qos, NULL);
+        CU_ASSERT_FATAL(topic1 > 0);
+
+        writer1 = dds_create_writer(participant, topic1, qos, NULL);
+        CU_ASSERT_FATAL(writer1 > 0);
+
+        reader1 = dds_create_reader(participant, topic1, qos, NULL);
+        CU_ASSERT_FATAL(reader1 > 0);
+
+        create_unique_topic_name("shared_memory", topicname, sizeof(topicname));
+        dds_entity_t topic2 = dds_create_topic(participant, &PsmxType1_desc, topicname, qos, NULL);
+        CU_ASSERT_FATAL(topic2 > 0);
+
+        writer2 = dds_create_writer(participant, topic2, qos, NULL);
+        CU_ASSERT_FATAL(writer2 > 0);
+
+        reader2 = dds_create_reader(participant, topic2, qos, NULL);
+        CU_ASSERT_FATAL(reader2 > 0);
+        dds_delete_qos(qos);
+      }
+      {
+        // Check that I get the same locator that I provided with the config.
+        ddsi_locator_t locator_out;
+        CU_ASSERT_FATAL(domain_get_psmx_locator(domain, &locator_out));
+        CU_ASSERT_FATAL(memcmp(locator_in, locator_out.address, sizeof(locator_in)) == 0);
+      }
+      {
+        // Check that shared memory is enabled.
+        bool psmx_enabled;
+        psmx_enabled = endpoint_has_psmx_enabled(writer1);
+        CU_ASSERT_FATAL(psmx_enabled);
+        CU_ASSERT_FATAL(dds_is_shared_memory_available(writer1));
+
+        psmx_enabled = endpoint_has_psmx_enabled(reader1);
+        CU_ASSERT_FATAL(psmx_enabled);
+        CU_ASSERT_FATAL(dds_is_shared_memory_available(reader1));
+
+        psmx_enabled = endpoint_has_psmx_enabled(writer2);
+        CU_ASSERT_FATAL(psmx_enabled);
+        CU_ASSERT_FATAL(dds_is_shared_memory_available(writer2));
+
+        psmx_enabled = endpoint_has_psmx_enabled(reader2);
+        CU_ASSERT_FATAL(psmx_enabled);
+        CU_ASSERT_FATAL(dds_is_shared_memory_available(reader2));
+      }
+      {
+        // Write and read samples
```

Comment:
I wouldn't include writing and reading samples in this test, because it's about testing the `is_shared_memory` flag. Maybe move this to a different test case?

Reply:
I thought as 'setting the flag' only is not enough and we should also see that it works as it should be so that we can read/write samples when shared memory is enabled. Putting in another test case is also an option but it means that we should repeat some beginning steps of this test case in that one, so I think keeping these steps here is easier for us.

Reply:
We have other tests (in this same source file) that do test communication over psmx, including over iox (so shared memory enabled). In fact, in certified, there is only PSMX communication and we only test over IOX, so all tests that do any communication test over shared memory. So I think we don't need to test that here.